### PR TITLE
protect against missing data

### PR DESCRIPTION
**Pull request by kevinkreiser**
if you have a partial tile set you need to watch out!

### CODE REVIEW

File: src/loki/search.cc [hunk 1]
```diff
@@ -207,8 +207,8 @@ PathLocation correlate_edge(GraphReader& reader, const Location& location, const
     //correlate its evil twin
     const GraphTile* other_tile;
     auto opposing_edge_id = reader.GetOpposingEdgeId(closest_edge_id, other_tile);
-    const auto* other_edge = other_tile->directededge(opposing_edge_id);
-    if(edge_filter(other_edge) != 0.0f) {
+    const DirectedEdge* other_edge;
+    if(opposing_edge_id.Is_Valid() && (other_edge = other_tile->directededge(opposing_edge_id)) && edge_filter(other_edge) != 0.0f) {
```
Comment (kevinkreiser) [inline src/loki/search.cc]: if you cant get the opposing edge you'll have to make due with just the one direction

File: src/loki/search.cc [hunk 2]
```diff
@@ -385,8 +385,11 @@ PathLocation search(const Location& location, GraphReader& reader, const EdgeFil
       bins += static_cast<size_t>(edges.size() > 0);
       for(auto e : edges) {
         //get the tile and edge
-        if(e.tileid() != tile->id().tileid())
+        if(!tile || e.tileid() != tile->id().tileid()) {
```
Comment (kevinkreiser) [inline src/loki/search.cc]: when getting opposing edges you may leave the tile for one that isnt there, protect against that

File: src/loki/search.cc [hunk 3]
```diff
@@ -424,12 +427,16 @@ PathLocation search(const Location& location, GraphReader& reader, const EdgeFil
   //it was the begin node
   if((front && closest_edge->forward()) || (back && !closest_edge->forward())) {
     const GraphTile* other_tile;
+    if(!other_tile)
```
Comment (kevinkreiser) [inline src/loki/search.cc]: if its the end of an edge we are going to have a hard time getting to the node if we can't leave the tile

**Comment by dnesbitt61** [comment]
:shipit:

**Pull request merged**